### PR TITLE
Built certifi for RHEL7.7 & RHEL8.0

### LINKED ISSUE #35: User Story: Setup discovery development environment for parallel development of RHEL 7 and 8 rpms
Currently, we have a system targeted at developing RPMs for either RHEL 7 or RHEL 8.  Spec files and other meta-data files are duplicated.  This story should create a system in which duplication is avoided.  The following requirements should be obtained:

- [x] Single spec file for both RHEL 7 and 8
- [x] Makefile that builds RHEL 7 or 8 
- [x] Vagrant files for both RHEL 7 and 8 combined which several VMs

### LINKED ISSUE #37: Build python-certifi for RHEL 7.7 & RHEL 8.0 in BREW

### PR DESCRIPTION
**Pull request by myersCody**
Closes #37 
Additionally starts in on #35 
We may not want to close 35 until all of the dependencies have been setup for duo development of rhel7.7 and rhel8

This PR sets up the development environment for both RHEL7 & RHEL8. In order to do that I needed to build an actually dependency as proof of concept. Therefore, I additionally built `python-certifi` for both RHEL7.7 & RHEL8.0

## Yum Repos:
*RHEL 8:* http://brew-task-repos.usersys.redhat.com/repos/scratch/cmyers/python-certifi/2018.10.15/4.el8d/python-certifi-2018.10.15-4.el8d-scratch.repo
*RHEL 7:* http://brew-task-repos.usersys.redhat.com/repos/scratch/cmyers/python-certifi/2018.10.15/4.el7d/python-certifi-2018.10.15-4.el7d-scratch.repo

Notice the new arch type `.el7d` & `.el8d`

## Tests:
**RHEL7:**
```
make test7
sudo su
cd /etc/yum.repos.d/
wget -O - http://brew-task-repos.usersys.redhat.com/repos/scratch/cmyers/python-certifi/2018.10.15/4.el7d/python-certifi-2018.10.15-4.el7d-scratch.repo >> python-certifi-2018.10.15-4.el7-scratch.repo
yum install python3-certifi
python3
Python 3.6.3 (default, Jan  9 2018, 10:19:07)
[GCC 4.8.5 20150623 (Red Hat 4.8.5-11)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import certifi
>>> certifi.where()
'/etc/pki/tls/certs/ca-bundle.crt'
```
**RHEL8:**
```
make test8
sudo su
cd /rcm_data/
./setup_rhel8_yum.sh
cd /etc/yum.repos.d/
yum install python3
wget -O - http://brew-task-repos.usersys.redhat.com/repos/scratch/cmyers/python-certifi/2018.10.15/4.el8d/python-certifi-2018.10.15-4.el8d-scratch.repo >> python-certifi-2018.10.15-4.el8d-scratch.repo
python3
Python 3.6.6 (default, Oct 16 2018, 01:53:53)
[GCC 8.2.1 20180905 (Red Hat 8.2.1-3)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import certifi
>>> certifi.where()
'/etc/pki/tls/certs/ca-bundle.crt'
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,36 @@
+DATE		= $(shell date)
+PYTHON		= $(shell which python)
+
+TOPDIR = $(shell pwd)
+
+
+
+help:
+	@echo "Please use \`make <target>' where <target> is one of:"
+	@echo "  setup-dev-7         Setup RHEL 7 VM for RHEL 7 dev"
```
Comment (kholdaway) [inline Makefile]: The help doesn't match the commands.

File: README.rst [hunk 2]
```diff
@@ -15,20 +15,149 @@ Development
...
+    make dev-7
+
+2. Setup rhpkg and brew. Note you will have to be connected to the VPN::
+
+    sudo su
     export KU=<YOUR_KERBEROS_USERNAME>
+    cd /rcm_data/deps/certifi/
+    make init-brew-7
+
+3. Modify the Makefile to build the dependency. For example, the following would need to be changed::
```
Comment (kholdaway) [inline README.rst]: We maybe should have more sections.  One for just building an existing component (not required to make changes).  Just explain how to do stuff.

Then another section for upgrading to a new version of Fedora (e.g. modify import to point to a new URL.  What would you expect to change (spec file, etc)

Then another section fo adding a new RMP.  Step 3 feels like that to me.

Comment (kholdaway) [reply README.rst]: Or just remove step 3 for now and state this is how you can build an existing dependency.  No changes to the makefile required.

File: CHEATSHEET.md [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+- Red Hat Container Registry: <https://docs.fedoraproject.org/en-US/packaging-guidelines/>
+- Distribute Content between products: <https://mojo.redhat.com/docs/DOC-1040116>
+- Open a ticket to link two products: <https://mojo.redhat.com/docs/DOC-1041766>
+##### Useful development links
+- Inspecting RPM packages: <https://blog.packagecloud.io/eng/2015/10/13/inspect-extract-contents-rpm-packages/>
+- Kerberos on your Mac (Google Chrome): <https://mojo.redhat.com/docs/DOC-968768>
+##### Mock Build Docs
+- Fedora Mockbuild: <https://fedoraproject.org/wiki/Using_Mock_to_test_package_builds>
+
+## Build System
```
Comment (abaiken) [inline CHEATSHEET.md]: Remove or update this section (out of date).

File: CHEATSHEET.md [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+##### Login to Kerberos
+`kinit -p $KU`
+
+##### How to clone a git-dist repository
+The `rhpkg co` command will clone any git dist package, as long as your user has access to it.
+
+```
+rhpkg co <git-dist>
+rhpkg co discovery
+rhpkg co discovery-server
```
Comment (abaiken) [inline CHEATSHEET.md]: we are phasing out discovery server so i would remove it here

File: README.rst [hunk 5]
```diff
@@ -10,25 +10,142 @@ discovery - Tool for discovery, inspection, collection, deduplication, and repor
 
 Development
 -----------
+It is recommended for new developers to start by reading the `RCM Cheatsheet. <https://github.com/redhat-discovery/discovery/blob/master/CHEATSHEAT>`_ It provides an introduction into RCM and the build system. Our development process uses Makefiles to automate the `Build System` section of the cheatsheet.
```
Comment (abaiken) [inline README.rst]: The `Build System` section was out of date, so if you remove it - i would remove this as well. If you update it, this statement will still work.

File: README.rst [hunk 6]
```diff
@@ -10,25 +10,142 @@ discovery - Tool for discovery, inspection, collection, deduplication, and repor
...
+
+4. Import a fedora rpm to the git-dist::
+
+    make import-8
+
+5. Test changes made to the spec file with a mock test::
+
+    make mock-8
+
+6. Launch a scratch build afer your mock build passes::
```
Comment (abaiken) [inline README.rst]: Fix `after`

File: README.rst [hunk 7]
```diff
@@ -10,25 +10,142 @@ discovery - Tool for discovery, inspection, collection, deduplication, and repor
...
 
 1. Clone the repository::
 
     git clone git@github.com:redhat-discovery/discovery.git
 
-2. Download `secret_setup_files.tar.gz file <https://drive.google.com/open?id=1t5DHIDIDov31pkjIM0KK6jka9Yi1EDqx>`_ and move to the directory ``path/to/discovery/rhel7``::
+2. Download `secret_setup_files.tar.gz file <https://drive.google.com/open?id=1t5DHIDIDov31pkjIM0KK6jka9Yi1EDqx>`_ and move to the directory ``path/to/discovery/``::
 
-3.  Bring up the vagrant machine and setup kerberos using the following commands::
+Developing for RHEL7
```
Comment (abaiken) [inline README.rst]: I feel like we don't need to split the sections up for 7/8. They are essentially the same except for `-8`/`-7` so we could find a way to document this more efficiently.

File: README.rst [hunk 8]
```diff
@@ -10,25 +10,142 @@ discovery - Tool for discovery, inspection, collection, deduplication, and repor
...
+
+4. Import a fedora rpm to the git-dist::
+
+    make import-7
+
+5. Test changes made to the spec file with a mock test::
+
+    make mock-7
+
+6. Launch a scratch build afer your mock build passes::
```
Comment (abaiken) [inline README.rst]: fix afer

File: README.rst [hunk 9]
```diff
@@ -10,25 +10,142 @@ discovery - Tool for discovery, inspection, collection, deduplication, and repor
...
+
+1. Create and access the Testing vagrant machine::
+
+    make test-8
+    sudo su
+    cd /rcm_data/
+    ./setup_rhel8_yum.sh
+    yum install -y python36
+
+2. Download yum repo. Each successful scratch build results an email, and the yum repo in the email will need to be saved in order to test the custom rpm.::
```
Comment (abaiken) [inline README.rst]: results in an email

File: rcm_data/deps/certifi/Makefile [hunk 10]
```diff
@@ -0,0 +1,127 @@
...
+RHEL7_DEV_DIR = rhel7dev
+RHEL8_DEV_DIR = rhel8dev
+
+help:
+	@echo "Please use \`make <target>' where <target> is one of:"
+	@echo "  (init-brew-7, init-brew-8)         Install BREW libs and authenticate"
+	@echo "  (setup-7, setup-8)                 Clone BREW git, checkout correct branch, and copy down build artifacts"
+	@echo "  (import-7, import-8)               Import Fedora RPM and copy up key files to directory tracked by git"
+	@echo "  (mock-7, mock-8)           		Copy down key files to brew git directory and run mock build"
+	@echo "  (scratch-7, mock-8)        		Copy down key files to brew git directory and run scratch build"
```
Comment (abaiken) [inline rcm_data/deps/certifi/Makefile]: (scratch-7, mock-8) ---> (scratch-7, scratch-8)

File: rcm_data/deps/certifi/Makefile [hunk 11]
```diff
@@ -0,0 +1,127 @@
...
+	@echo "  (build-7, mock-8)               	Copy down key files to brew git directory, cleanup mock build files, and print instructions to run a real build"
```
Comment (abaiken) [inline rcm_data/deps/certifi/Makefile]: (build-7, mock-8) ----> (build-7, build-8)

**Pull request merged**